File: c++/data_src_dependent/list_imdb/list_imdb.cpp
#include "Bridges.h"
#include "DataSource.h"
#include "Array.h"
#include "SLelement.h"


using namespace bridges;

int main(int argc, char **argv) {

	//create the Bridges object, set credentials
#if TESTING
                        // command line args provide credentials and server to test on
    Bridges bridges(atoi(argv[1]), argv[2], argv[3]);
    if (argc > 4)
        bridges.setServer(argv[4]);
#else
    Bridges bridges(YOUR_ASSSIGNMENT_NUMBER, "YOUR_USER_ID", 
                                "YOUR_API_KEY");
#endif
	
	bridges.setTitle("List with  IMDB Actor/Movie Data");

	DataSource ds (&bridges);
	std::vector< ActorMovieIMDB > ami = ds.getActorMovieIMDBData();

	//building linked list
	SLelement<ActorMovieIMDB>* head = nullptr;

	for (auto im : ami) {
	  SLelement<ActorMovieIMDB>* newone = new SLelement<ActorMovieIMDB> (im,
									     im.getActor() + " - " + im.getMovie());
	  newone->setNext(head);
	  head = newone;
	}
	
	// tell Bridges what data structure to visualize
	bridges.setDataStructure(head);

	// visualize the list
	bridges.visualize();

	//free memory
	while (head != nullptr) {
	  auto next = head->getNext();
	  delete head;
	  head = next;
	}
	
	return 0;
}

File: java/data_src_dependent/list_imdb/list_imdb.java
import java.util.List;

/* Import Bridges and relevant data source */
import bridges.connect.Bridges;
import bridges.base.SLelement;
import bridges.data_src_dependent.ActorMovieIMDB;

public class list_imdb {

	public static void main(String[] args) throws Exception {

		/* Initialize a Bridges connection with your credentials */
        Bridges bridges = new Bridges (Integer.parseInt(args[0]), args[1], args[2]);
        bridges.setServer (args[3]);

		/* Set an assignment title */
		bridges.setTitle("Linked  List using IMDB Actor/Movie Data");

		/* Get a List of ActorMovieIMDB objects from Bridges */
		List<ActorMovieIMDB> mylist = bridges.getActorMovieIMDBData(Integer.MAX_VALUE);

		/* Set up a prev and head element */
		SLelement<ActorMovieIMDB> prev = new SLelement<ActorMovieIMDB>();
		SLelement<ActorMovieIMDB> head = new SLelement<ActorMovieIMDB>();

		/* Read each actor movie pair and set up a new SLelement for each */
		for(int i = 0; i < mylist.size(); i++) {

			/* Create each new SLelement */
			SLelement<ActorMovieIMDB> element = new SLelement<ActorMovieIMDB>();

			/* Set the element label equal to 'Actor - Movie' for each pair. (print it out for sanity check) */
			element.setLabel(mylist.get(i).getActor() + " - " + mylist.get(i).getMovie());

			/* Add 'next' pointer where appropriate */
			if(i > 0) {
				prev.setNext(element);
			} else {
				/* Set the head pointer */
				head = element;
			}
			/* Update the prev pointer */
			prev = element;
		}

		/* Pass the head of the list to Bridges */
		bridges.setDataStructure(head);

		/* Visualize the list */
		bridges.visualize();
	}
}
